### ISSUE TITLE
Install devops and monitoring error in v1.25.9

### ISSUE THREAD

**Issue by littleBlackHouse** [kubesphere/kubesphere#5734]
**Describe the Bug**
when i install the plugins with devops and monitoring. the log in ks-installer appeared error: no matches for kind "PodDisruptionBudget" in version "policy/v1beta1".
the version of PodDisruptionBudget in v1.25.9 maybe policy/v1

For UI issues please also add a screenshot that shows the issue.

**Versions Used**
KubeSphere: v3.3.2
Kubernetes: v1.25.9

**Environment**
How many nodes and their hardware configuration:
Ubuntu Server 22.04 LTS 64-bit/ 1 masters and nodes   8cpu/16g; 1 nodes 4cpu/8g

**How To Reproduce**
1. change the configconfiguration cr to enable the plugin.
2. appear the error in ks-installer's log
<img width="1512" alt="截屏2023-05-25 18 05 04" src="https://github.com/kubesphere/kubesphere/assets/54946465/8d98c85d-f2b0-47cf-a75e-95ce179fc484">

**Expected behavior**

**Comment by luqiaoshi** [comment]
Issue fixed in `ks v3.4.0`
![image](https://github.com/kubesphere/kubesphere/assets/24789703/6cd9be42-885f-4f75-a794-03eeb0b578e3)

**Comment by krmao** [comment]
3.4.1 still exists
```
**************************************************
Waiting for all tasks to be completed ...
task network status is successful  (1/4)
task openpitrix status is successful  (2/4)
task multicluster status is successful  (3/4)
task monitoring status is failed  (4/4)
**************************************************
Collecting installation results ...


Task 'monitoring' failed:
******************************************************************************************************************************************************
{
  "counter": 148,
  "created": "2024-07-23T11:15:09.832850",
  "end_line": 141,
  "event": "runner_on_failed",
  "event_data": {
    "duration": 938.66661,
    "end": "2024-07-23T11:15:09.832734",
    "event_loop": null,
    "host": "localhost",
    "ignore_errors": null,
    "play": "localhost",
    "play_pattern": "localhost",
    "play_uuid": "a6219d20-9256-330f-dedb-000000000005",
    "playbook": "/kubesphere/playbooks/monitoring.yaml",
    "playbook_uuid": "315652d6-4591-4efd-bb0b-3940e7efe792",
    "remote_addr": "127.0.0.1",
    "res": {
      "_ansible_no_log": false,
      "attempts": 3,
      "changed": true,
      "cmd": "/usr/local/bin/helm upgrade --install notification-manager /kubesphere/kubesphere/notification-manager -f /kubesphere/kubesphere/notification-manager/custom-values-notification.yaml -n kubesphere-monitoring-system --force\n",
      "delta": "0:05:01.824416",
      "end": "2024-07-23 19:15:09.813365",
      "invocation": {
        "module_args": {
          "_raw_params": "/usr/local/bin/helm upgrade --install notification-manager /kubesphere/kubesphere/notification-manager -f /kubesphere/kubesphere/notification-manager/custom-values-notification.yaml -n kubesphere-monitoring-system --force\n",
          "_uses_shell": true,
          "argv": null,
          "chdir": null,
          "creates": null,
          "executable": null,
          "removes": null,
          "stdin": null,
          "stdin_add_newline": true,
          "strip_empty_ends": true,
          "warn": true
        }
      },
      "msg": "non-zero return code",
      "rc": 1,
      "start": "2024-07-23 19:10:07.988949",
      "stderr": "Error: UPGRADE FAILED: post-upgrade hooks failed: timed out waiting for the condition",
      "stderr_lines": [
        "Error: UPGRADE FAILED: post-upgrade hooks failed: timed out waiting for the condition"
      ],
      "stdout": "",
      "stdout_lines": []
    },
    "resolved_action": "command",
    "role": "ks-monitor",
    "start": "2024-07-23T10:59:31.166124",
    "task": "notification-manager | Deploying notification-manager",
    "task_action": "command",
    "task_args": "",
    "task_path": "/kubesphere/installer/roles/ks-monitor/tasks/notification-manager.yaml:30",
    "task_uuid": "a6219d20-9256-330f-dedb-00000000005a",
    "uuid": "410f3e37-90f3-4cae-8bec-bb0198d617e3"
  },
  "parent_uuid": "a6219d20-9256-330f-dedb-00000000005a",
  "pid": 6070,
  "runner_ident": "monitoring",
  "start_line": 140,
  "stdout": "fatal: [localhost]: FAILED! => {\"attempts\": 3, \"changed\": true, \"cmd\": \"/usr/local/bin/helm upgrade --install notification-manager /kubesphere/kubesphere/notification-manager -f /kubesphere/kubesphere/notification-manager/custom-values-notification.yaml -n kubesphere-monitoring-system --force\\n\", \"delta\": \"0:05:01.824416\", \"end\": \"2024-07-23 19:15:09.813365\", \"msg\": \"non-zero return code\", \"rc\": 1, \"start\": \"2024-07-23 19:10:07.988949\", \"stderr\": \"Error: UPGRADE FAILED: post-upgrade hooks failed: timed out waiting for the condition\", \"stderr_lines\": [\"Error: UPGRADE FAILED: post-upgrade hooks failed: timed out waiting for the condition\"], \"stdout\": \"\", \"stdout_lines\": []}",
  "uuid": "410f3e37-90f3-4cae-8bec-bb0198d617e3"
}
******************************************************************************************************************************************************
Failed to ansible-playbook result-info.yaml

```
